gtsfm/evaluation/visualize_benchmark_comparison.py

```diff
@@ -76,9 +76,15 @@ def plot_colored_table(row_labels: List[str], col_labels: List[str], tab_data: n
     Returns:
         string representing HTML code for the generated Plotly table.
     """
-
+    if tab_data.size == 0:
+        return ''
     # Clip "Z" to -20% and +20%. The clipping is only for the color -- the text will still display the correct numbers.
     tab_data_clipped = np.clip(tab_data, a_min=MIN_RENDERABLE_PERCENT_CHANGE, a_max=MAX_RENDERABLE_PERCENT_CHANGE)
+    hovertext = list()
+    for yi, yy in enumerate(col_labels):
+        hovertext.append(list())
+        for xi, xx in enumerate(row_labels):
+            hovertext[-1].append('Master_val: {}<br />Branch_val: {}<br />Percentage: {}'.format(old_values[yi][xi],new_values[yi][xi], tab_data[yi][xi]))
```

Comment:
If we're filling up a 2d table, I prefer we make it a bit more explicit, e.g.:
```python
>>> np.zeros((3,5), dtype=str).tolist()
[['', '', '', '', ''], ['', '', '', '', ''], ['', '', '', '', '']]
```
in our case,
```python
H, W = tab_data.shape
hovertext_table = np.zeros((H,W), dtype=str)
for r in range(H):
      for c in range(W):
             cell_text = f"Master: {old_values[r,c]}<br />Branch: {new_values[r,c} <br />Percentage: {tab_data[r,c}"
             hovertext_table[r, c] = cell_text
```
I'd prefer we use f-strings for readability, too.
Should the hover text table argument for Plotly be indexed by (y,x) or (x,y)?

Also curious how you found the documentation for this! Was there a page that was particularly helpful?

Reply:
i made these updates, should be good to go